include/picongpu/plugins/openPMD/openPMDWriter.hpp

```diff
@@ -386,6 +484,42 @@ make sure that environment variable OPENPMD_BP_BACKEND is not set to ADIOS1.
             std::string const description = "dump simulation data with openPMD";
             //! prefix used for command line arguments
             std::string const prefix = "openPMD";
+
+            std::vector<std::string> currentDataSources(uint32_t currentStep, size_t id)
+            {
+                if(tomlDataSources.has_value())
+                {
+                    return tomlDataSources.value().currentDataSources(currentStep);
+                }
+                else
+                {
+                    std::string dataSourceNames = source.get(id);
+                    return plugins::misc::splitString(plugins::misc::removeSpaces(dataSourceNames));
+                }
+            }
+
+            PluginOptions pluginOptions(size_t id)
+            {
+                if(tomlDataSources.has_value())
+                {
+                    return tomlDataSources.value().openPMDPluginOptions;
+                }
+                else
+                {
+                    PluginOptions res;
+                    for(auto const& param : parameters)
+                    {
+                        auto const& cmd_param = param.commandLineOption;
+                        auto const& target = param.targetInConfigObject;
+                        if(!target.has_value() || !cmd_param->optionDefined(id))
```

Comment:
Please review if the `!cmd_param->optionDefined(id)` condition is correct, especially when a default value is specified and not overwritten. This was so far only checked for the range option.